### ISSUE TITLE
[feat] support disable comment

### ISSUE THREAD

**Issue by tjx666** [johnsoncodehk/tsslint#12]
like eslint, we can disable by:

```ts
// tssint-disable-next-line
queryClient.invalidateQueries(['useXXX']);
```

**Comment by johnsoncodehk** [comment]
You can do this through the plugin API.

```ts
import { Plugin, defineConfig } from '@tsslint/config';

export default defineConfig({
	plugins: [
		createIngorePlugin(/\/\/ tssint-disable-next-line\n/g),
	]
});

function createIngorePlugin(pattern: RegExp): Plugin {
	return ({ languageService }) => ({
		resolveDiagnostics(fileName, results) {
			const sourceFile = languageService.getProgram()?.getSourceFile(fileName);
			if (!sourceFile) {
				return results;
			}
			const comments = [...sourceFile.text.matchAll(pattern)];
			const lines = new Set(comments.map(comment => sourceFile.getLineAndCharacterOfPosition(comment.index).line));
			return results.filter(result => !lines.has(sourceFile.getLineAndCharacterOfPosition(result.start).line - 1));
		},
	});
}
```

**Comment by tjx666** [comment]
@johnsoncodehk works fine, but I think this should be builtin feature.

**Comment by johnsoncodehk** [comment]
This is not yet a built-in feature, as projects migrating from ESLint may still need to support `eslint-disable-next-line` instead of `tsslint-disable-next-line`/`@tsslint-ignore`, so this syntax needs to be available configuration.

I will keep this issue open.

**Comment by tjx666** [comment]
@johnsoncodehk 

It would be fine to also support use tslint-disable-next-line, because some useful eslint rules which used to check unused eslint disable comments will report error to it:

<img width="1023" alt="image" src="https://github.com/user-attachments/assets/d605c544-7ff3-4bed-9b9d-c0d761ada568">

**Comment by tjx666** [comment]
for anyone don't know how to config the plugins, you can use like following:

<img width="871" alt="image" src="https://github.com/user-attachments/assets/a24fa40a-06f8-4b84-8958-7457b8291639">

**Comment by tjx666** [comment]
@johnsoncodehk the unused-disable-next-line check is not a good idea for eslint & tslint mixed project.

<img width="978" alt="image" src="https://github.com/user-attachments/assets/82f43cc1-c38d-4308-a476-a7b56896ed01">

**Comment by johnsoncodehk** [comment]
@tjx666 it can be disabled by `createDisableNextLinePlugin(false)`.

**Comment by tjx666** [comment]
@johnsoncodehk I see your latest implementation add 2 plugin arguments, so, I use following now(I just copy you plugin code because you hadnot pulish the latest code yet):

```
plugins: [
        createDisableNextLinePlugin(
            true,
            new RegExp(/\/\/\s*tslint-disable-next-line\b[ \t]*(?<ruleId>\S*)\b/g),
        ),
        createShowDocsActionPlugin(),
],
```

**Comment by johnsoncodehk** [comment]
> ```ts
> plugins: [
>         createDisableNextLinePlugin(
>             true,
>             new RegExp(/\/\/\s*tslint-disable-next-line\b[ \t]*(?<ruleId>\S*)\b/g),
>         ),
>         createShowDocsActionPlugin(),
> ],
> ```

This is now available in 1.3.0.
